errors/middleware-upgrade-guide.md

```diff
@@ -0,0 +1,290 @@
+# Middleware Upgrade Guide
+
+As we work on improving Middleware for General Availability (GA), we've made some changes to the Middleware APIs (and how you define Middleware in your application) based on your feedback.
+
+This upgrade guide will help you understand the changes and how to migrate your existing Middleware to the new API. This guide is for Next.js customers who:
+
+- Currently use the beta Next.js Middleware features
+- Choose to upgrade to the next stable version of Next.js
+
+> **Note:** For customers using Next.js on Vercel, your existing deploys using Middlware will continue to work. Existing sites can be redeployed without needing to upgrade. Before upgrading to the next stable version of Next.js, you will need to follow the upgrade guide below.
```

Comment:
I agree that we should be explicit about the version.

I wonder if we should pull this out into its own section instead of a note? As a developer, I'd want to understand what the immediate impact of these changes are to me first, and then understand how I adapt to these changes.

More text suggestions. 🙃 

```suggestion
## Using Next.js Middleware on Vercel

If you're using Next.js on Vercel, your existing deploys using Middleware will continue to work, and you can continue to deploy your site using Middleware. When you upgrade your site to the next stable version of Next.js, you will need to follow this upgrade guide to update your Middleware.
```